tools/strobelight/cli_function_profiler.py

```diff
@@ -0,0 +1,307 @@
+import functools
+import logging
+import os
+import re
+import subprocess
+import time
+from threading import Lock
+from typing import Any, List, Optional, Sequence
+
+# mypy: disallow-untyped-defs
+
+__all__ = [
+    "strobelight",
+    "StrobelightCLIProfilerError",
+    "StrobelightCLIFunctionProfiler",
+]
+
+logger = logging.getLogger("strobelight_function_profiler")
+
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "%(name)s, line %(lineno)d, %(asctime)s, %(levelname)s: %(message)s"
+)
+console_handler.setFormatter(formatter)
+
+logger.addHandler(console_handler)
+logger.setLevel(logging.INFO)
+logger.propagate = False
+
+
+class StrobelightCLIProfilerError(Exception):
+    """
+    Raised when an error happens during strobelight profiling
+    """
+
+
+def _pid_namespace_link(pid: Optional[int] = None) -> str:
+    PID_NAMESPACE_PATH = "/proc/{}/ns/pid"
+    """Returns the link to the process's namespace, example: pid:[4026531836]"""
```

Comment:
I think the docstring needs to be the first line of the function: https://peps.python.org/pep-0257/#what-is-a-docstring

Reply:
I copy pasted this function from an internal meta file i can move it